app/views/laboratory_samples/_form.html.haml

```diff
@@ -15,15 +15,96 @@
           = f.label :is_quality_control, "&nbsp;".html_safe
-
     - if @show_barcode_preview
       .col
         = render "barcode_card"
 
+  -# - if @sample.is_quality_control
+  = f.fields_for :test_qc_result, @sample.test_qc_result do |qc|
+    .row
+      .col.pe-2
+        = qc.label :test_qc_result_assays, "Assays"
+      .col
+        .choose-picture.assay
+          .upload-new-file.picture
+            -# .upload-picture-title
+            -#   %h2
+            -#     Drop a picture here
+            -#     %br or click to upload a file
+            = qc.fields_for :test_qc_result_assays do |ay|
+              .file-uploaded.picture-container
+                = image_tag ay.object.picture.url(:card)
+                .picture-title
+                  = ay.object.picture_file_name
+                .clear-label
+                  = ay.check_box '_destroy', { id: "destroy_assay_#{ay.object.id}" }
+                  %label.cross{:for => "destroy_assay_#{ay.object.id}" }
+                    = image_tag "ic-cross.png"
+
+            = qc.file_field :files, :class => "upload-picture",  accept: 'image/png,image/jpeg,image/gif', multiple: true
+
+
   .row.button-actions
     .col
       = f.submit 'Save', class: 'btn-primary'
       = link_to 'Cancel', edit_batch_path(@sample.batch), class: 'btn-link'
 
       - if @can_delete
         = confirm_deletion_button @sample, 'sample'
+
+:javascript
+  //custom_input_file(jQuery, window, document);
+
+  $(function() {
+    var onloadPicture = function(filename) {
+      return function(event) {
+        var fileDiv = $('<div>', { class: 'picture-container' })
+          .append(
+            $('<img>')
+              .attr('src', event.target.result)
+              .addClass('uploaded')
+          )
+          .append(
+            $('<div>', { class: 'picture-title' })
+              .text(filename)
+          );
+
+        $('.assay > .upload-new-file')
+          .append(fileDiv);
+      }
+    }
+
+    // Drag Picture to an input file and preview it
+    $('#laboratory_sample_test_qc_result_attributes_files').on('change', function(e) {
+      var files = e.target.files;
+
+      for(var i = 0; i < files.length; i++) {
+        var name = files[i].name;
+
+        var reader = new FileReader();
+        reader.onload = onloadPicture(name);
+        reader.readAsDataURL(files[i]);
+      }
+    });
+
+
+    // Change the class on the target zone when dragging a file over it
+    $(".clear-label").on('click', function (e) {
+      $(this).closest('.file-uploaded').addClass('remove');
+      e.stopPropagation();
+    });
+  });
+
+  document.addEventListener("dragenter", function( event ) {
+    if ( event.target.className == "upload-picture" ) {
+      $('.choose-picture').addClass('on');
+    }
+
+  }, false);
+
+  document.addEventListener("dragleave", function( event ) {
+    if ( event.target.className == "upload-picture" ) {
+      $('.choose-picture').removeClass('on');
+    }
+  }, false);
+
```

Comment:
Overall, code looks good. Maybe we should move this js code to a separate file, but only because I think that is too much js code to have in a `haml` file. Having it here only makes it larger and it does not help to code readability. What do you think?